File: sdk/python/omniutil.py
from uuid import uuid4
from datetime import datetime, UTC

def process_transaction(partner_id, user_id, amount_usd, reward_rate):
    return {
        "tx_id": str(uuid4()),
        "partner_id": partner_id,
        "user_id": user_id,
        "amount_usd": amount_usd,
        "merit_generated": amount_usd * reward_rate,
        "timestamp": datetime.now(UTC).isoformat()
    }

File: sdk/js/omniutil.js
export function processTransaction(partnerId, userId, amountUsd, rewardRate) {
  const merit = amountUsd * rewardRate;

  return {
    tx_id: crypto.randomUUID(),
    partner_id: partnerId,
    user_id: userId,
    amount_usd: amountUsd,
    merit_generated: merit,
    timestamp: new Date().toISOString()
  };
}
